packages/scripts/run-heavyweight-dedupe-test.sh

```
@@ -0,0 +1,119 @@
+#!/bin/bash
+
+# Check if base directory argument is provided
+if [ -z "$1" ]; then
+  echo "Error: Base directory path is required."
+  echo "Usage: $0 <base_directory>"
+  exit 1
+fi
+
+# Store base directory
+BASE_DIR="$1" 1>/dev/null 2>&1
+pushd $BASE_DIR 1>/dev/null 2>&1
```

Shellcheck (0.10.0)

[warning] Use 'pushd ... || exit' or 'pushd ... || return' in case pushd fails.

(SC2164)